plugins/woocommerce/templates/single-product/add-to-cart/grouped.php

```diff
@@ -64,7 +64,14 @@
 								woocommerce_template_loop_add_to_cart();
 							} elseif ( $grouped_product_child->is_sold_individually() ) {
 								echo '<input type="checkbox" name="' . esc_attr( 'quantity[' . $grouped_product_child->get_id() . ']' ) . '" value="1" class="wc-grouped-product-add-to-cart-checkbox" id="' . esc_attr( 'quantity-' . $grouped_product_child->get_id() ) . '" />';
-								echo '<label for="' . esc_attr( 'quantity-' . $grouped_product_child->get_id() ) . '" class="screen-reader-text">' . esc_html__( 'Buy one of this item', 'woocommerce' ) . '</label>';
+								echo '<label for="' . esc_attr( 'quantity-' . $grouped_product_child->get_id() ) . '" class="screen-reader-text">' .
+									sprintf(
+										/* translators: %1$s: Product name. %2$s: Product price. */
+										esc_html__( 'Buy one of %1$s for %2$s', 'woocommerce' ),
+										esc_html( $grouped_product_child->get_name() ),
+										esc_html( wp_strip_all_tags( $grouped_product_child->get_price_html() ) )
```

Comment:
When a product is `on sale` the regular+sale price info text appears concatenated which might be confusing:

<img width="623" alt="Screenshot 2025-02-17 at 13 01 51" src="https://github.com/user-attachments/assets/9ed62e02-caf9-4e9d-a72d-70eb3af8802b" />

Can we make sure this is formatted nicely?

Reply:
Great catch! I changed the output to catch on sale and regular prices in [ca5730d](https://github.com/woocommerce/woocommerce/pull/51964/commits/ca5730d8282d4142237299beccff7a477b368d72).

Reply:
Looks great, thanks!

<img width="1046" alt="Screenshot 2025-02-18 at 12 23 02" src="https://github.com/user-attachments/assets/e513928b-97e5-42a6-9a92-d6f708299d56" />